File: dotnet/Gherkin/StateMachine/StateMachineReader.cs
using System.Collections;
using System.Linq;
using System.Reflection;
using System.Resources;

namespace Gherkin.StateMachine
{
    using System;
    using System.Collections.Generic;
    using System.IO;
    using Lexer;

    public class StateMachineReader : IListener
    {
        private readonly string machinePath;
        private IList<IList<string>> transitionTable = new List<IList<string>>();

        public StateMachineReader(string name) {
            machinePath = "Gherkin.StateMachine." + name + ".txt";
        }

        public IList<IList<string>> TransitionTable 
        {
            get
            {
                ILexer lexer = new En(this);
                try
                {
                    var currentAssembly = Assembly.GetExecutingAssembly();
                    var stream = currentAssembly.GetManifestResourceStream(machinePath);
                    using (var reader = new StreamReader(stream))
                    {
                        lexer.Scan(reader);
                    }
                }
                catch (IOException)
                {
                    throw new Exception("Fatal error. Couldn't read " + machinePath);
                }
                return transitionTable;
            }
        }

        public void Tag(Token name)
        {
        }

        public void Comment(Token content)
        {
        }

        public void Feature(Token keyword, Token name)
        {
        }

        public void Background(Token keyword, Token name)
        {
        }

        public void Scenario(Token keyword, Token name)
        {
        }

        public void ScenarioOutline(Token keyword, Token name)
        {
        }

        public void Examples(Token keyword, Token name)
        {
        }

        public void Step(Token keyword, Token name, StepKind stepKind)
        {
        }

        public void Row(IList<Token> row, Position tablePosition)
        {
            transitionTable.Add(row.Select(cell => cell.Content).ToList());
        }

        public void PythonString(Token pyString)
        {
        }

        public void SyntaxError(string state, string @event, IEnumerable<string> legalEvents, Position position)
        {
        }
    }
}

File: java/src/main/java/gherkin/parser/StateMachineReader.java
package gherkin.parser;

import gherkin.lexer.Lexer;
import gherkin.lexer.Listener;
import gherkin.lexer.En;
import gherkin.util.FixJava;

import java.util.ArrayList;
import java.util.List;

public class StateMachineReader implements Listener {
    private final String machinePath;
    private List<List<String>> transitionTable;

    public StateMachineReader(String name) {
        machinePath = "/gherkin/parser/" + name + ".txt";
    }

    public List<List<String>> transitionTable() {
        transitionTable = new ArrayList<List<String>>();
        Lexer lexer = new En(this);
        lexer.scan(FixJava.readResource(machinePath));
        return transitionTable;
    }

    @Override
    public void tag(String tag, Integer line) {
    }

    @Override
    public void comment(String comment, Integer line) {
    }

    @Override
    public void feature(String keyword, String name, String description, Integer line) {
    }

    @Override
    public void background(String keyword, String name, String description, Integer line) {
    }

    @Override
    public void scenario(String keyword, String name, String description, Integer line) {
    }

    @Override
    public void scenarioOutline(String keyword, String name, String description, Integer line) {
    }

    @Override
    public void examples(String keyword, String name, String description, Integer line) {
    }

    @Override
    public void step(String keyword, String name, Integer line) {
    }

    @Override
    public void docString(String contentType, String content, Integer line) {
    }

    @Override
    public void eof() {
    }

    @Override
    public void syntaxError(String state, String event, List<String> legalEvents, String featureURI, Integer line) {
    }

    public void row(List<String> cells, Integer line) {
        transitionTable.add(cells);
    }
}
